### PR TITLE
test: Ensure commits not containing all enqueued proposals can be processed

### PR DESCRIPTION
**Pull request by augustocdias**

### CODE REVIEW

File: openmls/src/group/tests/test_commit_validation.rs [hunk 1]
```diff
@@ -672,3 +672,75 @@ fn test_valsem205(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
         .process_message(backend, MlsMessageIn::from(original_plaintext))
         .expect("Unexpected error.");
 }
+
+// this ensures that a member can process commits not containing all the stored proposals
+#[apply(ciphersuites_and_backends)]
+fn test_partial_proposal_commit(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
```
Comment (beltram) [inline openmls/src/group/tests/test_commit_validation.rs:678]: I've rewritten the test, there was some missing things.

```rust
// this ensures that a member can process commits not containing all the stored proposals
#[apply(ciphersuites_and_backends)]
fn test_partial_proposal_commit(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
    // Test with PublicMessage
    let CommitValidationTestSetup {
        mut alice_group,
        mut bob_group,
        ..
    } = validation_test_setup(PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);

    let charlie_index = alice_group
        .members()
        .find(|m| m.identity == b"Charlie")
        .unwrap()
        .index;

    // Create first proposal in Alice's group
    let proposal_1 = alice_group
        .propose_remove_member(backend, charlie_index)
        .unwrap();
    let proposal_1 = bob_group
        .process_message(backend, proposal_1.into())
        .unwrap();
    match proposal_1.into_content() {
        ProcessedMessageContent::ProposalMessage(p) => bob_group.store_pending_proposal(*p),
        _ => unreachable!(),
    }

    // Create second proposal in Alice's group
    let proposal_2 = alice_group.propose_self_update(backend, None).unwrap();
    let proposal_2 = bob_group
        .process_message(backend, proposal_2.into())
        .unwrap();
    match proposal_2.into_content() {
        ProcessedMessageContent::ProposalMessage(p) => bob_group.store_pending_proposal(*p),
        _ => unreachable!(),
    }

    // Alice creates a commit with only a subset of the epoch's proposals. Bob should still be able to process it.
    alice_group.proposal_store.pop().unwrap();
    let (commit, _) = alice_group.commit_to_pending_proposals(backend).unwrap();
    // Alice herself should be able to merge the commit
    alice_group
        .merge_pending_commit()
        .expect("Commits with partial proposals are not supported");

    // Bob should be able to process the commit
    bob_group
        .process_message(backend, commit.into())
        .expect("Commits with partial proposals are not supported");
    bob_group
        .merge_pending_commit()
        .expect("Commits with partial proposals are not supported");
}
```

File: openmls/src/group/tests/test_commit_validation.rs [hunk 2]
```diff
@@ -672,3 +672,75 @@ fn test_valsem205(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
...
+        .propose_add_member(backend, &dave_key_package)
+        .unwrap();
+    // Bob processes the proposal
+    process_proposal(&mut bob_group, msg_out);
+
+    // Alice makes a partial commit
+
+    // Drop one proposal from the queue so the commit includes only one proposal
+    // NOTE: there's no API to make a commit including selected proposals
+    alice_group.proposal_store.pop().unwrap();
```
Comment (beltram) [inline openmls/src/group/tests/test_commit_validation.rs:741]: I know it's painful but could you do the same without introducing this method ?

File: openmls/src/group/core_group/proposals.rs [hunk 3]
```diff
@@ -47,6 +47,11 @@ impl ProposalStore {
     pub(crate) fn empty(&mut self) {
         self.queued_proposals = Vec::new();
     }
+
+    #[cfg(test)]
+    pub(crate) fn pop(&mut self) -> Option<QueuedProposal> {
```
Comment (beltram) [inline openmls/src/group/core_group/proposals.rs:52]: remove this

**Review by beltram — CHANGES_REQUESTED** [review]

File: openmls/src/group/tests/test_commit_validation.rs (hunk 2 above)
Comment (augustocdias) [reply openmls/src/group/tests/test_commit_validation.rs]: can you check if this is better now?

**Pull request closed without merging**